source/grid/grid_in.cc

```diff
@@ -636,6 +636,82 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
                   this->cell_data[section_name] = std::move(temp_data);
                 }
             }
+
+          // Addition of POINT_DATA
+          else if (keyword == "POINT_DATA")
+            {
+              unsigned int n_ids;
+              in >> n_ids;
+
+              AssertThrow(n_ids == vertices.size(),
+                          ExcMessage(
+                            "The VTK reader found a POINT_DATA statement "
+                            "that lists a total of " +
+                            Utilities::int_to_string(n_ids) +
+                            " cell data objects, but this needs to "
+                            "equal the number of vertices (which is " +
+                            Utilities::int_to_string(vertices.size()) + ")"));
```

Comment:
Can VTK encode higher-order input?

Reply:
Yes it's possible, but it is not currently implemented in `read_vtk()`. For instance, `VTK_QUADRATIC_HEXAHEDRON` or `VTK_QUADRATIC_TETRA` are not supported.